.github/workflows/ubuntu.yml

```diff
@@ -109,125 +94,102 @@ jobs:
+        if: ${{ matrix.compiler-version.cc == 'gcc' }}
+        # We determined the dynamically linked libraries using ldd.
         run: |
           cd ../
-          tar cfz ${{ matrix.compiler-version }}.tar.gz downward lib
+          tar cfz archive.tar.gz downward/builds/debug/bin/ downward/builds/release/bin/ lib/coin/lib/libOsiCpx.so.1 lib/ibm/ILOG/CPLEX_Studio129/cplex/bin/x86-64_linux/libcplex1290.so lib/coin/lib/libOsiSpx.so.1 lib/coin/lib/libOsi.so.1 lib/coin/lib/libCoinUtils.so.3
 
-      - name: upload-files
-        if: ${{ matrix.compiler-version }} == gcc
-        uses: actions/upload-artifact@v1
+      - name: Upload archive
+        if: ${{ matrix.compiler-version.cc == 'gcc' }}
+        uses: actions/upload-artifact@master
         with:
           name: compiled-planner-${{ matrix.ubuntu-version }}
-          path: /home/runner/work/downward/${{ matrix.compiler-version }}.tar.gz
+          path: /home/runner/work/downward/archive.tar.gz
 
 
   test:
-    name: test
+    name: Test planner
     runs-on: ${{ matrix.ubuntu-version }}
     needs: compile # TODO: this only depends on the compile step with gcc
     strategy:
       matrix:
         ubuntu-version: [ubuntu-18.04, ubuntu-20.04]
-        compiler-version: [gcc]
         python-version: [3.6, 3.8]
         exclude:
           - ubuntu-version: ubuntu-18.04
             python-version: 3.8
     steps:
-      - name: setup-python
-        uses: actions/setup-python@v1
+      - name: Install Python
+        uses: actions/setup-python@master
         with:
           python-version: ${{ matrix.python-version }}
 
-      - name: download-files
-        uses: actions/download-artifact@v1
+      - name: Install dependencies
+        run: |
+          pip3 install tox
+          sudo apt-get -y install zlib1g-dev libgmp3-dev gcc flex bison
+
+        # NOTE: VAL does not compile with clang-11.
+      - name: Install VAL
+        run: |
+          git clone https://github.com/KCL-Planning/VAL.git
+          cd VAL
+          git checkout a5565396007eee73ac36527fbf904142b3077c74
+          make clean  # Remove old build artifacts and binaries.
+          sed -i 's/-Werror //g' Makefile  # Ignore warnings.
+          make -j2
+          mv validate ../
+          cd ../
+          rm -rf VAL
+
+      - name: Download archive
+        uses: actions/download-artifact@master
         with:
           name: compiled-planner-${{ matrix.ubuntu-version }}
 
-      - name: extract-files
-        # We need to make sure that paths are the same as in the first job,
-        # otherwise cmake exits with an error when called during tests.
-        # Alternatively, we could change the tests so that they don't build.
+      - name: Extract archive
+        # We need to make sure that paths are the same as in the first
+        # job, otherwise cmake exits with an error when called during
+        # tests. Alternatively, we could change the tests so that they
+        # don't build.
```

Comment:
This should be resolved now, right?

Reply:
We need to get the library paths right. I updated the comment.